bindings/java/src/org/sleuthkit/datamodel/Image.java

```diff
@@ -539,6 +598,23 @@ public String getAcquisitionDetails() throws TskCoreException {
 		return getSleuthkitCase().getAcquisitionDetails(this);
 	}	
 
+	/**
+	 * Updates the image's total size and sector size.This function may be used
+	 * to update the sizes after the image was created.
+	 *
+	 * Can only update the sizes if they were not set before. Will throw
+	 * TskCoreException if the values in the db are not 0 prior to this call.
+	 *
+	 * @param totalSize  The total size
+	 * @param sectorSize The sector size
+	 *
+	 * @throws TskCoreException Thrown if the values were not 0 in DB
```

Comment:
Why is it necessary to enforce this restriction? Is seems like we are putting application/business logic in the data layer and there is a classical check-then-act race condition in the underlying SleuthkitCase method (see comment there).

Reply:
The theory was that we only want someone to change the size if it was not set when the image was initially created.  So, the 0 check was to prevent any random module from updating the settings. I think it is a good safety check to minimize changes that any module can make. 

We have similar logic elsewhere.